doc/source/reference/io.rst

```diff
@@ -25,6 +25,19 @@ Flat file
    DataFrame.to_csv
    read_fwf
 
+.. currentmodule:: pandas.io.parsers
+
+.. autosummary::
+   :toctree: api/
+
+   TextFileReader
+
+   TextFileReader.get_chunk
```

Comment:
We obviously don't want to add these. But when I remove them, sphinx complains that it can not find them in any toctree. Anyone any ideas how to solve this? For ExcelWriter below this works, so I am probably missing something.

Reply:
I think in the past we used a section with `:hidden:`, like in https://github.com/pandas-dev/pandas/blame/main/doc/source/getting_started/index.rst#L641

But I don't see it being used for the API anymore. I guess we should just make them private by using `_get_chunk`... if we don't want them public and in the documentation.

Or maybe just `_TextFileReader` and make the whole class private if we don't want it being part of our public API.

I'm personally fine with any of them.

Reply:
I think we could deprecate it and remove it later, that sounds good to me.

We can't makeTextFileReader private, since it is returned by read_csv if you are reading the file in chunks

Reply:
Ah, true. Sounds good to me then. Also find to simply document the methods and publish them in the docs. Maybe we can just add a not for now that we recommend using the magic methods way.